core/init.go

```diff
@@ -139,83 +152,115 @@ func initOauth2ForService(config *tomlConfig, url string, c *http.Client) {
 		s := fmt.Sprintf("Failed to set up oauth2 authentication with error %s.", err.Error())
 		lc.Error(s)
 	} else {
+		defer resp.Body.Close()
 		if resp.StatusCode == 200 || resp.StatusCode == 201 || resp.StatusCode == 409 {
 			lc.Info("Successful to set up oauth2 authentication.")
 		} else {
 			s := fmt.Sprintf("Failed to set up oauth2 authentication with errorcode %d.", resp.StatusCode)
 			lc.Error(s)
 		}
 	}
+}
 
+func initAuthmethod(config *tomlConfig, url string, c *http.Client) {
+	lc.Info(fmt.Sprintf("selected auth method as %s.", config.KongAuth.Name))
+	if config.KongAuth.Name == "jwt" {
```

Comment:
a `switch` here might be nice if we anticipate adding support for other auth methods

Reply:
OK I will get it into the place for the next PR.